File: content/_code-samples/checks/py/cancel_check.py
from xrpl.wallet import generate_faucet_wallet
from xrpl.clients import JsonRpcClient
from xrpl.models import CheckCancel
from xrpl.transaction import submit_and_wait

client = JsonRpcClient("https://s.altnet.rippletest.net:51234") # Connect to the testnetwork

"""Cancel a check"""
# Sender is the check creator or recipient
# If the Check has expired, any address can cancel it

# Check id
check_id = "F944CB379DEE18EFDA7A58A4F81AF1A98C46E54A8B9F2D268F1E26610BC0EB03"

# Create wallet object
sender_wallet = generate_faucet_wallet(client=client)

# Build check cancel transaction
check_txn = CheckCancel(account=sender_wallet.address, check_id=check_id)

# Sign and submit transaction
stxn_response = submit_and_wait(check_txn, client, sender_wallet)

# Parse response for result
stxn_result = stxn_response.result

# Print result and transaction hash
print(stxn_result["meta"]["TransactionResult"])
print(stxn_result["hash"])

File: content/_code-samples/checks/js/cancel_check.js
const xrpl = require('xrpl');

const secret = "sEdTPPEeMH6SAgpo6rSj8YW7a9vFfUj";
const checkId = "505E0A4CCBCC971EE07DCD25611A612830CFFA4D50DBC04947EF952D898C94F3"; // Replace with your check ID

const main = async ()=> {
  try {
      
    // Connect ----------------------------------------------------------------
    const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233');
    await client.connect();

    // Prepare ----------------------------------------------------------------
    const wallet = await xrpl.Wallet.fromSeed(secret);
    const checkCancelRequest = {
        "TransactionType": "CheckCancel",
        "Account": wallet.address,
        "CheckID": checkId
    };

    // Auto-fill the fields ---------------------------------------------------
    const prepared = await client.autofill(request);

    // Submit -----------------------------------------------------------------
    const response = await client.submitAndWait(prepared, { wallet });
    console.log(JSON.stringify(response.result, null, "\t"));

    // Disconnect -------------------------------------------------------------
    await client.disconnect();

  } catch (error) {
      console.error(`Error: ${error}`);
  }
}

main();